hyper/tls.py

```diff
@@ -31,7 +31,7 @@ def wrap_socket(sock, server_hostname):
     global _context
 
     if _context is None:  # pragma: no cover
-        _context = _init_context()
+        _context = ssl_context or _init_context()
```

Comment:
I actually think this logic is wrong. The provided `ssl_context` should be used only for the specific call to wrap_socket, but this affects all subsequent calls. I think probably we should use a different, local variable in this method, and then prefer `ssl_context` to `_context`.